### ISSUE TITLE
log_api() function working unexpectly

### ISSUE THREAD

**Issue by yoonack** [cuckoosandbox/cuckoo#792]
I tried this sample: f2c093d16bea877fdd40dc7880762ef6 to debug what went wrong with the monitor. I managed to trace down two situations, one unexpectly close at log_stacktrace function and the other one in log_buffer function. 

I download the version from monitor and make a debug version. The fault occurs at calling API LdrLoadDll that tried to load shell32.dll to get address for shellexecute. I found that program terminated inside the log_stacktrace functions at code: 
              bson_append_string(b, number, sym);

I tried bypassing calling this log_debug, the second fault occurs right inside the function NtWriteFile, however, the same error would occur inside NtWriteVirtualMemory too. I traced inside, and found that the error happened to fall in the log_buffer function tried to log the buffer data with length, this is the code that happens to generate the error:
    if(range_is_readable(buf, length) != 0) {
        bson_append_binary(b, idx, BSON_BIN_BINARY,
            (const char *) buf, trunclength);                    //this function causes fault
    }

I thought it could be something related to mongoDB, so I upgraded mongoDB to the newest version, however, nothing is solved. Someone has better idea about these errors?

**Comment by jbremer** [comment]
It's getting stuck in a `GetCursorPos`/`NtDelayExecution` loop for me :(

**Comment by yoonack** [comment]
Oh, I think this one the malware is used to trap the automated system, however, looks like mine is able to automatically bypass this one.

**Comment by yoonack** [comment]
Does the same situation happen to you? @jbremer

**Comment by jbremer** [comment]
I don't think it bypassed the mouse position thing, no. So that probably needs some extra work..

**Comment by yoonack** [comment]
hmm...thats weird, for me, this is the API log that it automatically changed cursor:
March 1, 2016, 2:26 p.m.
GetCursorPos	y: 425 
x: 43 
SUCCESS	1	0
March 1, 2016, 2:26 p.m.
NtDelayExecution
skipped: 0 
milliseconds: 1 
SUCCESS	0	0
March 1, 2016, 2:26 p.m.
GetCursorPos	y: 425 
x: 43 
SUCCESS	1	0
March 1, 2016, 2:26 p.m.
NtDelayExecution
skipped: 0 
milliseconds: 1 
SUCCESS	0	0
March 1, 2016, 2:26 p.m.
GetCursorPos	y: 425 
x: 43 
SUCCESS	1	0
March 1, 2016, 2:26 p.m.
NtDelayExecution
skipped: 0 
milliseconds: 1 
SUCCESS	0	0
March 1, 2016, 2:26 p.m.
GetCursorPos	y: 425 
x: 43 
SUCCESS	1	0
March 1, 2016, 2:26 p.m.
NtDelayExecution	
skipped: 0 
milliseconds: 1 
SUCCESS	0	0
March 1, 2016, 2:26 p.m.
GetCursorPos	y: 0 
x: 927

**Comment by jbremer** [comment]
So you're saying that after a couple of these calls the sample starts behaving *normally* for you?
If it keeps going on like this then your analysis isn't very useful :)

**Comment by yoonack** [comment]
This loop of GetCursorPos is just normal part of this sample. My machine is managed to bypass just using the latest monitor. However, if it keeps going, as I described above, there are two places that would cause the program to terminate unexpectly instead of behaving as it is in OllyDBG.

**Comment by yoonack** [comment]
@jbremer have you tried running in OllyDBG? This seems like something to do with MongoDB, but I don't know how to fix that. Thanks for the help.
